vault/custom_response_headers.go

```diff
@@ -0,0 +1,193 @@
+package vault
+
+import (
+	"fmt"
+	log "github.com/hashicorp/go-hclog"
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"github.com/hashicorp/vault/internalshared/configutil"
+)
+
+type ListenersCustomResponseHeadersList struct {
+	CustomHeadersList []*ListenerCustomHeaders
+}
+
+type ListenerCustomHeaders struct {
+	Address string
```

Comment:
Looks like these fields could all be private, since they're only referenced in methods.